src/components/ingestion/AbstractIngestionEngine.js

```diff
@@ -5,61 +5,90 @@ import {
 	AbstractInstantiationError,
 	NotImplementedError,
 } from '@tvkitchen/base-errors'
+import { TSDemuxer } from 'ts-demuxer'
+
 import logger from '%src/lib/logger'
 import kafka from '%src/lib/kafka'
+import { tsToSeconds } from '%src/lib/utils/mpegts'
 
 class AbstractIngestionEngine {
 	constructor() {
 		if (this.constructor === AbstractIngestionEngine) {
 			throw new AbstractInstantiationError(this.constructor.name)
 		}
 		this.producer = kafka.producer()
+		this.mpegtsDemuxer = new TSDemuxer(this.processDemuxedPacket)
+		this.ffmpegProcess = null
+		this.currentPts = 0
 	}
 
 	/**
-	 * Ingests data into the TV Kitchen pipeline.
+	 * Ingests a mpegts payload into the TV Kitchen pipeline.
 	 *
-	 * @param  {Buffer} data The buffer of data to be ingested.
-	 * @return {Boolean}     Whether or not the buffer was successfully ingested.
+	 * @param  {Payload} data The payload to be ingested.
+	 * @return {Boolean}      Whether or not the payload was successfully ingested.
 	 */
-	ingestData = async (data) => {
-		const payload = new Payload()
-		payload.data = data
+	ingestPayload = async (payload) => {
+		if (!(payload instanceof Payload)) {
+			throw new Error('ingestPayload must be passed a Payload object')
+		}
 		await this.producer.send({
-			topic: dataTypes.STREAM.DATA,
+			topic: dataTypes.STREAM.CONTAINER,
 			messages: [{
-				value: data,
+				value: Payload.serialize(payload),
 			}],
 		})
 		return true
 	}
 
 	/**
-	 * Continuously ingest of a stream of data.
+	 * Updates ingestion state based on the latest mpegts packet data as
+	 * parsed by our demuxer.
 	 *
-	 * @param  {ReadableStream} readableStream The stream of data to be recorded.
-	 * @return {ReadableStream}                The input stream (for chaining)
+	 * @param  {Packet} packet A Packet object, as defined by the TSDemuxer project
 	 */
-	recordStream = (readableStream) => readableStream
-		.on('data', async (data) => {
-			readableStream.pause()
-			await this.ingestData(data)
-			readableStream.resume()
+	processDemuxedPacket = (packet) => {
+		this.currentPts = packet.pts
+	}
+
+	/**
+	 * Process a new chunk of data from an mpegts stream.
+	 * This data is used to update local state and ultimately pipe to kafka.
+	 *
+	 * @param  {Buffer} mpegtsData The latest arbitrary chunk of buffered mpegts data.
```

Comment:
I think it's worth clarifying that this is a chunk of arbitrary _size_, not, like, arbitrary composition or order.